internal/planner/planner.go

```diff
@@ -85,6 +88,85 @@ func GeneratePlan(
 	return plan, nil
 }
 
+// checkAndUpdateDeploymentConnectionSpec checks if a deployment needs to be updated. This is true if the connection spec it's using is outdated.
+func checkAndUpdateDeploymentConnectionSpec(
+	versionID string,
+	k8sState *k8s.DeploymentState,
+	connection temporaliov1alpha1.TemporalConnectionSpec,
+) *appsv1.Deployment {
+	existingDeployment, exists := k8sState.Deployments[versionID]
+	if !exists {
+		return nil
+	}
+
+	// If the connection spec hash has changed, update the deployment
+	currentHash := k8s.ComputeConnectionSpecHash(connection)
+	if currentHash != existingDeployment.Spec.Template.Annotations[k8s.ConnectionSpecHashAnnotation] {
```

Comment:
is it possible for annotations to ever be nil? like if the user doesn't provide any or something. Could you make sure to handle the case where the user provides no initial annotations?